### PR TITLE
Set the Google::Apis.logger so it doesn't default to Rails.logger

### PR DESCRIPTION
**Pull request by agrare**
Depends on: https://github.com/ManageIQ/manageiq/pull/20235

Fixes https://github.com/ManageIQ/manageiq-providers-google/issues/140

### CODE REVIEW

File: app/models/manageiq/providers/google/manager_mixin.rb [hunk 1]
```diff
@@ -77,6 +77,9 @@ def verify_credentials(args)
     end
 
     def raw_connect(google_project, google_json_key, options, proxy_uri = nil, validate = false)
+      require "google/apis"
+      ::Google::Apis.logger = $gce_log
```
Comment (agrare) [inline app/models/manageiq/providers/google/manager_mixin.rb:81]: @Fryguy not sure if there's a better place to do this at startup without eager requiring `"google/apis"`

**Comment by djberg96** [comment]
Looks ok to me, the codeclimate complaint isn't that severe.

File: app/models/manageiq/providers/google/manager_mixin.rb (hunk 1 above)
Comment (djberg96) [reply app/models/manageiq/providers/google/manager_mixin.rb:81]: This is basically what we do in the other providers, too, isn't it?

Comment (agrare) [reply app/models/manageiq/providers/google/manager_mixin.rb:81]: Well its not ideal that we can't set the logger on the fog::compute object and that it is a "global" setting but not much we can do about that, that's just how fog-google and goolgle-api-client works

**Pull request merged**